File: Tests/Main Tests/ProjectTestsCSharp/TestAssembly.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Tests.Core;

namespace ProjectTestsCSharp
{
    public class TestAssembly : ITestAssembly
    {
        private List<ITestPackage> _listPackages;

        internal static TestAssembly Singleton { get; private set; }

        public TestAssembly()
        {
            Singleton = this;
        }

        #region ITestAssembly Members

        public string Language
        {
            get { return "C#"; }
        }

        public string OfficeProduct
        {
            get { return "Project"; }
        }

        public ITestPackage[] LoadTestPackages()
        {
            if (null == _listPackages)
            {
                _listPackages = new List<ITestPackage>();
                _listPackages.Add(new Test01());
                _listPackages.Add(new Test02());

            }
            return _listPackages.ToArray();
        }

        #endregion
    }
}

File: Tests/Main Tests/ProjectTestsVB/TestAssembly.vb
Imports Tests.Core

Public Class TestAssembly
    Implements ITestAssembly

    Private _listPackages As List(Of ITestPackage)

    Public ReadOnly Property Language As String Implements Tests.Core.ITestAssembly.Language
        Get
            Return "VB"
        End Get
    End Property

    Public Function LoadTestPackages() As Tests.Core.ITestPackage() Implements Tests.Core.ITestAssembly.LoadTestPackages

        If IsNothing(_listPackages) Then

            _listPackages = New List(Of ITestPackage)
            _listPackages.Add(New Test01())
            _listPackages.Add(New Test02())

        End If

        Return _listPackages.ToArray()

    End Function

    Public ReadOnly Property OfficeProduct As String Implements Tests.Core.ITestAssembly.OfficeProduct
        Get
            Return "Project"
        End Get
    End Property

End Class
